service/src/main/java/org/databiosphere/workspacedataservice/dataimport/AvroRecordConverter.java

```diff
@@ -185,6 +188,40 @@ public Object convertAttributeType(Object attribute) {
     return attribute.toString();
   }
 
+  // Parquet has a concept of "physical type" and "logical type". For some numeric types,
+  // the physical type is a fixed-size byte array (GenericData.Fixed), but the logical type
+  // defines the desired shape of the value, such as Decimal. This method converts the fixed-size
+  // byte arrays to decimals.
+  private Object convertFixed(GenericData.Fixed fixedAttr) {
+    // check if this fixed has a logical type
+    LogicalType logicalType = fixedAttr.getSchema().getLogicalType();
+
+    // handle logical decimals
+    if (logicalType instanceof LogicalTypes.Decimal) {
+      // transform the bytes into a BigDecimal. This BigDecimal will have scale and precision
+      // specified by the schema, which can be much larger than the actual values and leads
+      // to trailing zeros in the decimal part of the number
+      BigDecimal bigDecimal =
+          new Conversions.DecimalConversion()
+              .fromFixed(fixedAttr, fixedAttr.getSchema(), logicalType);
+
+      // now, try to turn the BigDecimal back into an int or a double; this truncates the
+      // precision to the actual value. Example: 0.220000000 will become 0.22
+      Object returnValue;
+      try {
+        returnValue = bigDecimal.intValueExact();
+      } catch (ArithmeticException ae) {
+        returnValue = bigDecimal.doubleValue();
+      }
+      return returnValue;
```

Comment:
Up to you, but slightly more concise as...?

```suggestion
      try {
        return bigDecimal.intValueExact();
      } catch (ArithmeticException ae) {
        return bigDecimal.doubleValue();
      }
```

Reply:
ah, good call, my code was kinda sloppy! fixed.